packages/node_modules/@webex/http-core/src/request/request.shim.js

```diff
@@ -93,11 +94,23 @@ export default function _request(options) {
    */
   function bindProgressEvents(params, o) {
     if (params.method && ['PATCH', 'POST', 'PUT'].includes(params.method.toUpperCase())) {
-      params.xhr = new XMLHttpRequest();
+      if (!params.xhr) {
```

Comment:
When would there be the case that `xhr` wouldn't exist at this point if the only place this private function is called is right after `setXhr`?

Reply:
it's for completeness and modularity. Bind progress events OR setXHR could disappear at any time, and we want the function to be complete on it's own.